### PR TITLE
Enhancement | Light / Dark Mode 

### LINKED ISSUE #150: Feature | Dark / Light mode
Setup Dark & Light mode for the app where light mode = currrent setup. 

The Color, typography, spacing, roundedness, etc. should be easily configurable from 1 config file.
Build using tailwind CSS.

Ensure that you're implementing modes in the most optimised, efficient manner.

### **DO NOT BREAK EXISTING FEATURES**

### PR DESCRIPTION
**Pull request by vverma022**
# Pull Request Title

Dark/Light Mode Implementation

## Description

Implements Dark mode for the website with the use of Shadcn UI. Also makes Appbar responsive when viewing in smaller sized screens.

Resolves #150 

## Linked Issues

- Fixes #[Issue Number]
- Closes #159 

## Type of Change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Changes

- Added custom text and background utility classes which can be used in the future for designing the ui without writing css colour properties.
- Implemented a dark/light mode toggle using Shadcn ui which lets you switch between 3 options Dark,Light and System default.
- Changed the Appbar ti make it responsive when using the website in a smaller screen

## Screenshots/Recordings

Screenshot 2024-06-30 at 11 17 37 AM

https://github.com/aadeshkulkarni/figuringout/assets/109036913/18242c20-fba4-4448-804e-166d34e001fe

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I assure there are no similar/duplicate pull requests regarding the same issue
- [x] My changes follow the project's coding guidelines and best practices
- [x] Code is formatted properly and lint check passes successfully
- [ ] I have made corresponding changes to the documentation (if applicable)

### CODE REVIEW

**Comment by aadeshkulkarni** [comment]
@vverma022 - npm build fails

**Comment by vverma022** [comment]
> @vverma022 - npm build fails

I have some dependencies not installed ?

**Comment by aadeshkulkarni** [comment]
> > @vverma022 - npm build fails
> 
> I have some dependencies not installed ?

You need to run the `npm run build` command on the frontend project to check & fix errors.

**Comment by aadeshkulkarni** [comment]
1. The border color for dark mode is very light. It's creating a bad UX. Can you please keep the border to border-slate-800 from border-gray-200 ?
2. Dark mode: The default text color can be a darker shade across the app. text-gray-100/200 / text-slate-100/200

File: frontend/src/index.css [hunk 1]
```diff
@@ -1,194 +1,104 @@
...
-    }
-    body {
-      overflow-y: scroll; 
-      scroll-snap-type: y mandatory;
-    }
-    .snap-center {
-      scroll-snap-align: center;
-    }
-  }
-}
```
Comment (aadeshkulkarni) [inline frontend/src/index.css:18]: Why have we removed these classes ?

File: frontend/src/index.css [hunk 2]
```diff
@@ -1,194 +1,104 @@
-@import url('https://fonts.googleapis.com/css2?family=Noto+Sans:ital,wdth,wght@0,62.5..100,100..900;1,62.5..100,100..900&display=swap');
```
Comment (aadeshkulkarni) [inline frontend/src/index.css:1]: Why is this removed ?

Please verify all your changes. Make sure you are not removing any essential code.

**Comment by vverma022** [comment]
> 1. The border color for dark mode is very light. It's creating a bad UX. Can you please keep the border to border-slate-800 from border-gray-200 ?
> 
> 2. Dark mode: The default text color can be a darker shade across the app. text-gray-100/200 / text-slate-100/200

Alright will implement these changes

**Review by aadeshkulkarni — APPROVED** [review]
